operator/v1/types_openshiftservicecatalogapiserver.go

```diff
@@ -0,0 +1,43 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OpenShiftServiceCatalogAPIServer provides information to configure an operator to manage Service Catalog API Server
+type OpenShiftServiceCatalogAPIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OpenShiftServiceCatalogAPIServerSpec   `json:"spec,omitempty"`
+	Status OpenShiftServiceCatalogAPIServerStatus `json:"status,omitempty"`
+}
+
+type OpenShiftServiceCatalogAPIServerSpec struct {
+	OperatorSpec `json:",inline"`
+
+	//enabled specifies if Service Catalog should be installed and started in the cluster
+	Enabled bool `json:"enabled"`
+
+	//serverArguments are key value pairs that will override the default set of arguments.  ServerArguments may include Feature Gates & Admission Plugins.  If you specify an invalid argument the server will not start. These values may cause invalid configurations.",
+	ServerArguments map[string][]string `json:"serverArguments"`
```

Comment:
Why do we need to expose this?

Reply:
initially added to enable cluster admins to configure with different sets of feature flags and provide flexibility.  Discussed with Derek and agree it provides to much flexibility, if needed specific feature gates should be exposed.  Alternatively, admins could go on their own and set to non managed/unsupportedConfigOverride